sentry/src/main/java/io/sentry/adapters/DoubleDeserializerAdapter.java

```diff
@@ -0,0 +1,50 @@
+package io.sentry.adapters;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+import io.sentry.DateUtils;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.util.Objects;
+import java.lang.reflect.Type;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class DoubleDeserializerAdapter implements JsonDeserializer<Double> {
```

Comment:
does that mean that this class and `DoubleSerializerAdapter` will be triggered for every Double in the protocol classes and if its not related to the Date/Double changes?

Reply:
Yes, and I think there's no easy way to set up an adapter only for fields related to one type - we should I think have a separate serializer for `SentryTransaction` and `SentrySpan`. Perhaps its better to wait for the non-gson serializer to be merged till we merge this one?

Reply:
yeah I'd say so, so we can target the ser/deser correctly